#!/usr/bin/env python
PACKAGE = "cnbiros_shared_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Repellors orientation parameters
gen.add("repellors_strength",	double_t, 0, "Strength of repellors", 0.5, 0, 10)
gen.add("repellors_decay",		double_t, 0, "Distance decay of repellors", 1.0, 0, 10)
gen.add("repellors_occupancy",		double_t, 0, "Occupancy of repellors", 1.0, 0, 10)

# Attractors parameters
gen.add("command_timeout", double_t, 0, "Timeout for user commands",	8.0, 0, 20)

# Goal distance parameter
gen.add("distance_max", double_t, 0, "Distance maximum",	2.0, 0.0, 10)
gen.add("distance_slope", double_t, 0, "Distance slope",	1.3, 0.1, 3.0)
gen.add("distance_zero", double_t, 0, "Distance zero",		1.2, 0.0, 10)

# Autonomous goal generation 
gen.add("reset_radius", double_t, 0, "Radius of reset area",	1.4, 0.1, 3.0)

# Node parameters
gen.add("update_rate",	double_t, 0, "Node update rate", 10.0, 0.1, 20)

exit(gen.generate(PACKAGE, "shared_actions_parameters", "SharedActions"))
